addOns/spider/src/test/java/org/zaproxy/addon/spider/parser/DsStoreParserUnitTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.spider.parser;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.BDDMockito.given;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.jupiter.api.Test;
+
+/** Unit test for {@link DsStoreParser}. */
+class DsStoreParserUnitTest extends SpiderParserTestUtils<DsStoreParser> {
+
+    private static final Path BASE_DIR_TEST_FILES =
+            getResourcePath(DsStoreParserUnitTest.class, "dsstore");
+
+    @Override
+    protected DsStoreParser createParser() {
+        try {
+            msg.setRequestHeader("GET / HTTP/1.1\r\nHost: example.com\r\n");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        given(spiderOptions.isParseDsStore()).willReturn(true);
+        return new DsStoreParser();
+    }
+
+    @Test
+    void shouldFailToEvaluateAnUndefinedContext() {
+        // Given
+        ParseContext ctx = null;
+        // When / Then
+        assertThrows(NullPointerException.class, () -> parser.canParseResource(ctx, false));
+    }
+
+    @Test
+    void shouldFailToEvaluateAnUndefinedPath() {
+        // Given
+        given(ctx.getPath()).willReturn(null);
+        // When / Then
+        assertThrows(NullPointerException.class, () -> parser.canParseResource(ctx, false));
+    }
+
+    @Test
+    void shouldParseMessageEvenIfAlreadyParsed() {
+        // Given
+        boolean parsed = true;
+        given(ctx.getPath()).willReturn("/.DS_Store");
+        // When
+        boolean canParse = parser.canParseResource(ctx, parsed);
+        // Then
+        assertThat(canParse, is(equalTo(true)));
+    }
```

Comment:
I modeled this on the sitemap unit tests. This along with a few others don't appear to actually use `msg` so I'm not sure that it's actually of any value. Deciding whether something should be parsed or not (in the .DS_Store case anyway) depends on more than just path.